pkg/syncclient/sync_client.go

```diff
@@ -324,6 +355,13 @@ func (s *SyncerClient) loop(cxt context.Context, cancelFn context.CancelFunc) {
 		case syncproto.MsgServerHello:
 			logCxt.WithField("serverVersion", msg.Version).Info("Server hello message received")
 
+			// Check whether Typha supports node resource updates.
+			if !msg.SupportsNodeResourceUpdates {
+				logCxt.Info("Server responded without support for node resource updates, assuming older Typha")
+			}
+			s.supportsNodeResourceUpdates = msg.SupportsNodeResourceUpdates
+			s.handshakeStatus.helloReceivedChan <- struct{}{}
```

Comment:
A bit marginal, but I wonder if it's nicer to pass `msg.SupportsNodeResourceUpdates` (and potentially, in future, any other flags that are determined by the handshake) through the channel, instead of writing a struct field on this goroutine and reading it on another.